hs_core/models.py

```diff
@@ -4108,6 +4136,11 @@ def show_in_discover(self):
         if any([res.raccess.discoverable for res in replaced_by_resources]):
             # there is a newer discoverable resource - so this resource should not be shown in discover
             return False
+
+        if not self.spam_allowlisted:
+            if not self.free_of_spam_patterns:
```

Comment:
the `show_in_discover` property is used during SOLR update in deployment. In my testing, the regex search takes .01 sec per resource so I'm hoping this doesn't slow down our deployments substantially